### PR TITLE
Removed .env configuration

### PR DESCRIPTION
**Pull request by nicosomb**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Removed .env configuration
| Type?             | improvement 
| BC breaks?        | no
| Deprecations?     | no
| Sponsor company   | PrestaShop SA

This PR allows to fix the issue I have here https://github.com/PrestaShop/PrestaShop/pull/36198#issuecomment-2122042886 to embed HB in PrestaShop core.

### CODE REVIEW

**Comment by nicosomb** [comment]
I know that some variables will remain empty, but the build seems ok like this. 

Removing .env dependency will help us to ship hummingbird in prestashop core.

**Review by Hlavtox — CHANGES_REQUESTED** [review]
The env configuration was added for a reason. 🤔 Ping @Oksydan @NeOMakinG

**Comment by nicosomb** [comment]
I know that there were some reasons ... but we can't find them.

**Comment by Hlavtox** [comment]
PR that introduced this, with discussion - https://github.com/PrestaShop/hummingbird/pull/67

**Comment by nicosomb** [comment]
OK so it seems that you warned about this issue...

**Comment by Oksydan** [comment]
@nicosomb .env file is being used by webpack-dev-server mostly to improve DX. Why would you like to get ride of this like this 🤔.

**Comment by nicosomb** [comment]
@Oksydan I understand, but can you tell us how can we ship it in PrestaShop core please ? 

We want to add Hummingbird as a dependency in composer.json (see my PR here https://github.com/PrestaShop/PrestaShop/pull/36198). So we have to know how to fill the .env file during the assets build process.

**Comment by Oksydan** [comment]
Hmmmmm we could pass arguments in the compilation process. If the process is building a theme, we could simply not check if the `.env` file exists, otherwise check for its existence and terminate the process. For missing values from env, we could set default values:
```javascript 
const {
  PORT: port = null,
  PUBLIC_PATH: publicPath = null,
  SERVER_ADDRESS: serverAddress = null,
  SITE_URL: siteURL = null,
} = process.env;
```
THB we do not need these values in the compilation process. It is only required for the development process.

**Comment by nicosomb** [comment]
@Oksydan thank you for the details. I will improve my PR.

File: templates/catalog/_partials/miniatures/product.tpl [hunk 1]
```diff
@@ -176,7 +176,7 @@
               {/block}
             </div>
 
-            {if $product.add_to_cart_url && $product.product_type !== 'combinations'}
+            {if $product.add_to_cart_url}
```
[the remark this answers is not in the document]
Comment (boherm) [reply templates/catalog/_partials/miniatures/product.tpl:179]: This is related to this PR: https://github.com/prestashop/hummingbird/pull/609/

[the remark this answers is not in the document]
Comment (nicosomb) [reply templates/catalog/_partials/miniatures/product.tpl:179]: I know that. But it seems that it breaks UI tests.

[the remark this answers is not in the document]
Comment (boherm) [reply templates/catalog/_partials/miniatures/product.tpl:179]: I think that's an issue about UI tests then.
If we merge your diff, we reintroduce the "Add to cart" button for combinations products, and his buggy behaviour in that particular case 🤔

But anyway, I don't block your PR, just I prefer to warn you about this! 😉

**Comment by nicosomb** [comment]
@Hlavtox we have to go forward, could you please remove your request approval? 
if necessary, we can still open a new PR later.

File: templates/catalog/_partials/miniatures/product.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hlavtox) [reply templates/catalog/_partials/miniatures/product.tpl:179]: I think the change is legit, add to cart url should be handled on product lazy array settings. For combination, it's availability depends on prestashop settings (enable add to cart button in listings when product has combinations on/off).

**Review by Hlavtox — APPROVED** [review]

**Comment by florine2623** [comment]
Hello @nicosomb ,

What are the steps to reproduce ?
Do we just need to make sure the theme is well installed ?

**Review by florine2623 — APPROVED** [review]
Hello @nicosomb ,

Hummingbird is well installed.
Made an order with all types of products, order process is working as expected.

LGTM ✅ 
Thanks!

**Pull request merged**